R/summary.pop_data.R

```diff
@@ -12,21 +13,40 @@
 #' library(dplyr)
 #'
 #' xs_data <- load_pop_data("https://osf.io/download//n6cp3/")
+#' summary(xs_data, strata = "Country")
 #'
-#' summary(xs_data)
-#'
-summary.pop_data <- function(object, ...) {
+summary.pop_data <- function(object, strata = "Country", ...) {
+  # get relevant columns from object
+  age_column <- object %>% get_age_var()
+  value_column <- object %>% get_value_var()
+  id_column <- object %>% get_id_var()
+
   ages <-
     object %>%
-    distinct(.data$id, .data$age)
+    distinct(
+      .data[[age_column]],
+      .data[[id_column]],
+      .data[[strata]]
+    )
 
   cat("\nn =", nrow(ages), "\n")
 
   cat("\nDistribution of age: \n\n")
   age_summary <-
     ages %>%
-    pull("age") %>%
-    summary() %>%
+    select(
+      age_column,
+      all_of(strata)
+    ) %>%
+    group_by(across(all_of(strata))) %>%
```

Comment:
please avoid using `group_by()` unless the grouping needs to persist across multiple steps; otherwise use the `.by` argument in `summarise()`, `mutate()`, etc.

Reply:
I have made the change.